src/igrep/igrep.cpp

```diff
@@ -178,33 +179,34 @@ main (int argc, const char *argv[])
                 "-a", &all_subimages, "Search all subimages of each file",
                 "--help", &help, "Print help message",
                 NULL);
+    // clang-format off
     if (ap.parse(argc, argv) < 0 || pattern.empty() || filenames.empty()) {
         std::cerr << ap.geterror() << std::endl;
         ap.usage ();
         return EXIT_FAILURE;
     }
     if (help) {
-        ap.usage ();
-        exit (EXIT_FAILURE);
+        ap.usage();
+        exit(EXIT_FAILURE);
     }
 
 #if USE_BOOST_REGEX
-    boost::regex_constants::syntax_option_type flag = boost::regex_constants::grep;
+    boost::regex_constants::syntax_option_type flag
+        = boost::regex_constants::grep;
     if (extended_regex)
         flag = boost::regex::extended;
     if (ignore_case)
         flag |= boost::regex_constants::icase;
 #else
-    std::regex_constants::syntax_option_type flag = std::regex_constants::grep;
+    auto flag = std::regex_constants::grep;
```

Comment:
clang-format is also able to translate types to `auto`? Or did you run `clang-tidy` as well?

Reply:
No, that was me. :-)

As I reviewed the diffs to see what it rewrote and why, in a *very* small number of cases of long lines or awkward constructs, where the reformatted code looked especially ugly, I occasionally made a change or two of this nature to coax it into a form that didn't need to re-wrap in a weird way.

Reply:
Makes sense. LGTM!

Reply:
I'm pretty happy about the whole thing overall. The deltas were really extensive, but 90% was totally neutral, stuff like "foo (bar)" -> "foo(bar)" and the like.  Some changes are different than I would have written them and I think they can be improved, but they are within the bounds of what I would accept from a team member contributing to the project, and I think worth accepting my occasional peeves with its choices for the peace of mind of having all future contributors able to trivially 100% conform to a single formatting standard and allow it to be checked automatically.